### ISSUE TITLE
[rllib] Custom metrics compatibility with MultiAgentEnv

### ISSUE THREAD

**Issue by sanjeevanahilan** [ray-project/ray#5669]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu
- **Ray version**: 0.6.1
- **Python version**: 3.6.8

### Describe the problem

The examples show how to build custom metrics for single agent environments, which involves the assignment `episode = info["episode"]`. See example code: https://github.com/ray-project/ray/blob/master/rllib/examples/custom_metrics_and_callbacks.py

However, following this straightforwardly does not work for MultiAgentEnv e.g. multiagent_cartpole.py, as I get `KeyError: 'episode'`. I would like custom metrics for each agent in the multi-agent environment, as well as metrics for the population of agents (this is separate from metrics for each policy). What's the best way to do this? (apologies if I missed something obvious).

**Comment by ericl** [comment]
Could you provide a reproduction script? I think it should work actually.

**Comment by kiddyboots216** [comment]
Hi -this is a working example of creating metrics in an `on_train_result` callback function:
https://github.com/fedpbt/fedpbt/blob/master/fed/utils.py#L200
Hopefully this can clear up how to access the metrics you want.

**Comment by sanjeevanahilan** [comment]
Hi, thanks for your responses and the link. Turns out I actually made a stupid mistake and it does work fine - sorry about that.

I will close this issue, although I have another question in case you happen to know the answer. Is it possible to create custom metrics for the MultiAgentEnv rather than just the agents experience? Right now, it seems as though 'episode' only gives access to the observations of the agents rather than the true environment state. In my case each of the agents does not see the full state, but I would like to compute metrics for that state.
